pkg/endpoint/policy.go

```diff
@@ -240,6 +240,32 @@ func (e *Endpoint) regenerateConsumable(owner Owner) (bool, error) {
 	return true, nil
 }
 
+// Must be called with endpointsMU held
+func (e *Endpoint) regenerateL3Policy(owner Owner) (bool, error) {
+	c := e.Consumable
+
+	c.Mutex.RLock()
+	ctx := policy.SearchContext{
+		To:    c.LabelList,
```

Comment:
I'm having some issues with all the `DeepCopy()`. While it avoids changes under a live pointer, it does not solve the race itself. After the Unlock, the LabelList can change and we calculate policy for stale labels without knowing it.

I agree with the `DeepCopy()` fix short term but we need to look at this very closely and make sure we are providing accurate guarantees regarding the policy that is generated.